File: core/web/console-fe/src/model/builder/SqlColumn.ts
import {SqlOrder} from "@/model/builder/SqlOrder";

export class SqlColumn
{
  column: string;
  order: SqlOrder;

  constructor(builder: SqlColumnBuilder)
  {
    this.column = builder.column;
    this.order = builder.order;
  }
}

export class SqlColumnBuilder
{
  private readonly _column: string;
  private _order: SqlOrder;

  constructor(column: string)
  {
    this._column = column;
  }

  setOrder(order: SqlOrder)
  {
    this._order = order;
    return this;
  }

  get column()
  {
    return this._column;
  }

  get order()
  {
    return this._order;
  }

  build()
  {
    return new SqlColumn(this);
  }
}

File: core/datacap-common/src/main/java/io/edurt/datacap/common/sql/configure/SqlColumn.java
package io.edurt.datacap.common.sql.configure;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Properties;

@Data
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@SuppressFBWarnings(value = {"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
public class SqlColumn
{
    private String column;
    private String value;
    private String expression;
    private String alias;
    private SqlOperator operator;
    private SqlOrder order;
    private Properties original;
}
